docs/Reference-for-template.json.md

```diff
@@ -408,3 +408,30 @@ Primary outputs define the list of template files for further processing.
 |condition|if the condition evaluates to `true`, the corresponding path will be added to primary outputs, if `false`, the path is ignored. If no condition is provided for a path, the condition defaults to `true`.|
 
 For more information on primary outputs, refer to [the article](https://github.com/dotnet/templating/wiki/Using-Primary-Outputs-for-Post-Actions).
+
+
+### Template constraints
+
+Available since .NET SDK 7.0.100.
+The template may define the constraints which should be met in order that the template can be used. 
```

Comment:
As non-native speaker I might be totally wrong here :-), but I feel small corrections are needed, plus I'd make it explict that all constraints must be met.

E.g. (but might be off):

```
The template may define constraints, all of which must be met in order for the template to be instanciable and visible in `list` command output.
```

Reply:
I would like to avoid `list` mention, because it also applied to Visual Studio. Hence `to be used`. It's up to host to decide how constraints are used.